syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package proto_catalogue;

message Coordinates
{
    double lat = 1;
    double lng = 2;
}

message DistanceToStops
{
    int32 stop_to_id = 1;
    int32 distance = 2;
}

message Stop
{
    string name = 1;
    int32 id = 2;
    Coordinates coordinates = 3;   
    repeated DistanceToStops distance_to_stops = 4;
}

message Bus
{
    string name = 1;
    bool is_circular = 2;
    repeated int32 stops = 3;
}

message TransportCatalogue
{
    repeated Stop stops = 1;
    repeated Bus buses = 2;
}

message TransportNavigator 
{
    TransportCatalogue catalogue = 1;
    proto_map_renderer.RenderSettings render_settings = 2;
    proto_tr_router.TransportRouter transport_router = 3;
}